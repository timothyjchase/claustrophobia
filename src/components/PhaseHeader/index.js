import React from 'react'
import PropTypes from 'prop-types'
import { Step } from 'semantic-ui-react'

const PHASES = [
  { key: 'INITIATIVE', text: 'Initiative' },
  { key: 'HUMAN_ACTION', text: 'Human', text2: 'Actions' },
  { key: 'THREAT', text: 'Threat' },
  { key: 'DEMON_ACTION', text: 'Demon', text2: 'Actions' },
]

const PhaseHeader = ({ phase }) => (
  <Step.Group unstackable fluid>
    {PHASES.map(phaseConfig => (
      <Step
        key={phaseConfig.key}
        active={phase === phaseConfig.key}
        disabled={phase !== phaseConfig.key}
        style={{ padding: '5px 0px 5px 0px' }}
      >
        <Step.Content>
          <Step.Description>{phaseConfig.text}</Step.Description>
          {!!phaseConfig.text2 && (
            <Step.Description>{phaseConfig.text2}</Step.Description>
          )}
        </Step.Content>
      </Step>
    ))}
  </Step.Group>
)

PhaseHeader.propTypes = {
  phase: PropTypes.string.isRequired,
}

export default PhaseHeader
