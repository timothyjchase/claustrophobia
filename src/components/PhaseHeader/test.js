import React from 'react'
import PhaseHeader from './index'

describe('<PhaseHeader />', () => {
  describe('with phase = INITIATIVE', () => {
    const component = shallow(<PhaseHeader phase="INITIATIVE" />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with phase = HUMAN_ACTION', () => {
    const component = shallow(<PhaseHeader phase="HUMAN_ACTION" />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with phase = THREAT', () => {
    const component = shallow(<PhaseHeader phase="THREAT" />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with phase = DEMON_ACTION', () => {
    const component = shallow(<PhaseHeader phase="DEMON_ACTION" />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
