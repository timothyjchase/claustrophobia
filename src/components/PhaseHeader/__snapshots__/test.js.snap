// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhaseHeader /> with phase = DEMON_ACTION should match snapshot 1`] = `
<StepGroup
  fluid={true}
  unstackable={true}
>
  <Step
    active={false}
    disabled={true}
    key="INITIATIVE"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Initiative
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={false}
    disabled={true}
    key="HUMAN_ACTION"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Human
      </StepDescription>
      <StepDescription>
        Actions
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={false}
    disabled={true}
    key="THREAT"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Threat
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={true}
    disabled={false}
    key="DEMON_ACTION"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Demon
      </StepDescription>
      <StepDescription>
        Actions
      </StepDescription>
    </StepContent>
  </Step>
</StepGroup>
`;

exports[`<PhaseHeader /> with phase = HUMAN_ACTION should match snapshot 1`] = `
<StepGroup
  fluid={true}
  unstackable={true}
>
  <Step
    active={false}
    disabled={true}
    key="INITIATIVE"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Initiative
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={true}
    disabled={false}
    key="HUMAN_ACTION"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Human
      </StepDescription>
      <StepDescription>
        Actions
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={false}
    disabled={true}
    key="THREAT"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Threat
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={false}
    disabled={true}
    key="DEMON_ACTION"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Demon
      </StepDescription>
      <StepDescription>
        Actions
      </StepDescription>
    </StepContent>
  </Step>
</StepGroup>
`;

exports[`<PhaseHeader /> with phase = INITIATIVE should match snapshot 1`] = `
<StepGroup
  fluid={true}
  unstackable={true}
>
  <Step
    active={true}
    disabled={false}
    key="INITIATIVE"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Initiative
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={false}
    disabled={true}
    key="HUMAN_ACTION"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Human
      </StepDescription>
      <StepDescription>
        Actions
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={false}
    disabled={true}
    key="THREAT"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Threat
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={false}
    disabled={true}
    key="DEMON_ACTION"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Demon
      </StepDescription>
      <StepDescription>
        Actions
      </StepDescription>
    </StepContent>
  </Step>
</StepGroup>
`;

exports[`<PhaseHeader /> with phase = THREAT should match snapshot 1`] = `
<StepGroup
  fluid={true}
  unstackable={true}
>
  <Step
    active={false}
    disabled={true}
    key="INITIATIVE"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Initiative
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={false}
    disabled={true}
    key="HUMAN_ACTION"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Human
      </StepDescription>
      <StepDescription>
        Actions
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={true}
    disabled={false}
    key="THREAT"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Threat
      </StepDescription>
    </StepContent>
  </Step>
  <Step
    active={false}
    disabled={true}
    key="DEMON_ACTION"
    style={
      Object {
        "padding": "5px 0px 5px 0px",
      }
    }
  >
    <StepContent>
      <StepDescription>
        Demon
      </StepDescription>
      <StepDescription>
        Actions
      </StepDescription>
    </StepContent>
  </Step>
</StepGroup>
`;
